todos-app/src/App/App.tsx

```diff
@@ -1,18 +1,49 @@
-import React, { useState } from "react";
+import { useEffect, useState } from "react";
 import NewTodoInput from "./NewTodoInput/NewTodoInput";
 import TodoList from "./TodoList/TodoList";
 import ListFooter from "./ListFooter/ListFooter";
-import { ITodo } from "../DataStructure";
+import { Todo } from "./models/Todos";
+import { FilterType } from "./models/Filters";
 import "../App.css";
 
-const App: React.FC = () => {
-  const [todos, setTodos] = useState<ITodo[]>([]);
+const App = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterTodos, setFilterTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<FilterType>("All");
 
-  const addHandler = (input: ITodo) => {
+  useEffect(() => {
+    setFilterTodos(todos);
+    switch (filter) {
+      case "All": {
+        break;
+      }
+      case "Active": {
+        setFilterTodos((prev) =>
+          prev.filter((todo) => todo.completed === false)
+        );
+        break;
+      }
+      case "Completed": {
+        setFilterTodos((prev) =>
+          prev.filter((todo) => todo.completed === true)
+        );
+        break;
+      }
+
+      default:
+        break;
+    }
+  }, [todos, filter]);
+
+  const filterTodosHandler = (filterValue: FilterType) => {
+    setFilter(filterValue);
+  };
+
+  const addHandler = (input: Todo) => {
     setTodos((prev) => [...prev, input]);
   };
 
-  const toggleHandler = (id: string) => {
+  const toggleTodoHandler = (id: string) => {
```

Comment:
`toggleTodoCompleted`

Reply:
I use this handler to toggle and untoggle my todos. Therefore, I think the name _toggleTodoCompleted_ doesn't fully reflect the essence.